src/meta.c

```diff
@@ -68,6 +68,16 @@ void broadcast_meta(meshlink_handle_t *mesh, connection_t *from, const char *buf
 		}
 }
 
+void broadcast_submesh_meta(meshlink_handle_t *mesh, connection_t *from, submesh_t *s,
+                            const char *buffer, int length) {
+	for list_each(connection_t, c, mesh->connections)
+		if(c != from && c->status.active) {
+			if(c->node && (!c->node->submesh || (!s || (s == c->node->submesh)))) {
```

Comment:
That's a lot of parentheses, I think it's clearer if you write:

`if(c->node && (!c->node->submesh || !s || s == c->node->submesh))`

Actually, since we do this kind of comparison in more places, it might make sense to add a function in `submesh.c` that checks if a node is allowed to communicate with the given submesh, so that here you could write comething like:

`if(c->node && submesh_allows_node(s, c->node))`

Reply:
I made this condition check a different function in submesh.c as you suggested